persistence/src/main/scala/akka/persistence/LocalActorRef.scala

```diff
@@ -0,0 +1,140 @@
+package akka.persistence
+
+import akka.actor.ActorRef
+import akka.actor.MinimalActorRef
+import cats.effect.Temporal
+import cats.effect.syntax.all._
+import cats.syntax.all._
+import com.evolutiongaming.catshelper.CatsHelper.OpsCatsHelper
+import com.evolutiongaming.catshelper.SerialRef
+import com.evolutiongaming.catshelper.ToTry
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import java.util.concurrent.TimeoutException
+import scala.concurrent.duration.FiniteDuration
+
+/** Representation of actor capable of constructing result from multiple messages passed into the actor. Inspired by [[PromiseActorRef]] but
+  * result [[R]] is an aggregate from incomming messages rather that first message. Can be used only locally, does _not_ tolerate.
+  * [[ActorRef.provider]] and [[ActorRef.path]] functions.
+  * @tparam F
+  *   The effect type.
+  * @tparam R
+  *   The result type of the aggregate.
+  */
+private[persistence] trait LocalActorRef[F[_], R] {
+
+  def ref: ActorRef
```

Comment:
It might be worth to mention here or in trait scaladoc, that this is not an actual actor, but just an implementation of `ActorRef` abstract class, so it could be passed instead of a persistentActor to Akka journal plugin. :thinking: